ReactSkia/platform/linux/RuntimeEventBeat.cpp

```diff
@@ -5,23 +5,67 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+#include <thread>
+
 #include "RuntimeEventBeat.h"
 #include "ReactSkia/utils/RnsLog.h"
 
+#define BEAT_INTERVAL  1 /*Temp:1 sec*/
+
 namespace facebook {
 namespace react {
 
-RuntimeEventBeat::RuntimeEventBeat(EventBeat::SharedOwnerBox const &ownerBox, RuntimeExecutor runtimeExecutor)
-    : EventBeat(ownerBox), runtimeExecutor_(std::move(runtimeExecutor)) {
-    RNS_LOG_NOT_IMPL;
+namespace {
+
+/*
+EventBeatThread : to push the dispatched asynchronous event to the JS world
+TODO: BEAT_INTERVAL to be refined to the need.
+*/
+void BeatThread(RuntimeEventBeat *runtimeEventBeat,RunLoopObserver::Activity activities) {
+  while(RuntimeEventBeat::keepBeating) {
+    sleep(BEAT_INTERVAL);
+    runtimeEventBeat->triggerBeat(activities);
+  }
+}
+
+}//namespace
+
+bool RuntimeEventBeat::keepBeating{true};
+/*Activity is dummy param to fit in React common implementation, It is not needed as
+  RNS implementation is not based on runloop*/
+RunLoopObserver::Activity activities{RunLoopObserver::Activity::BeforeWaiting};
+RuntimeEventBeat::RuntimeEventBeat(RunLoopObserver::WeakOwner const &owner)
+  : RunLoopObserver(activities, owner) {
+  /*Spawning  Beat Thread*/
+  std::thread thread(BeatThread,this,activities);
```

Comment:
Can we use folly::ScopedEventBaseThread here and schedule event at every 1s ?
Reference : TimingModule() in JSITurboModuleManager.
This will avoid using additional bool and sleep here 

Also because most of the threads which we have created are using folly lib;